pkg/registry/navigator/escluster/strategy.go

```diff
@@ -97,14 +99,8 @@ type esClusterStatusStrategy struct {
 }
 
 func (esClusterStatusStrategy) PrepareForUpdate(ctx genericapirequest.Context, new, old runtime.Object) {
-	newESCluster, ok := new.(*navigator.ElasticsearchCluster)
-	if !ok {
-		glog.Fatal("received a non-elasticsearchcluster object to update to")
-	}
-	oldESCluster, ok := old.(*navigator.ElasticsearchCluster)
-	if !ok {
-		glog.Fatal("received a non-elasticsearchcluster object to update from")
-	}
+	newESCluster := new.(*navigator.ElasticsearchCluster)
+	oldESCluster := old.(*navigator.ElasticsearchCluster)
```

Comment:
I guess it's safe to assume that it'll always be supplied with the right types? Or is this only because of the added validation in this branch?

Reply:
My reasoning here: previously we called Fatal (which calls os.Exit). Now we call panic. It was always going to bring everything down, we've just changed how now and tidied things up. FWIW, yes we should always be passed the correct type. Upstream just directly casts (like this change makes us do), so figured we should do similar :)